### PR TITLE
LPD-22758 Fix NewEnvJVMTestRule so that New JVM would use the same java as the current environment, not the system default

### PR DESCRIPTION
**Pull request by julschong**
@tinatian per our discussion,
This is the part that isolated from the property writing/reading from ant or gradle.

We are only adding test and fixing NewEnv to make sure that when junit.jvm.override property is set:
1. NewEnv's java.home == Junit's java.home
2. NewEnv JAVA_HOME == Junit's JAVA_HOME

### CODE REVIEW

**Comment by julschong** [comment]
ci:test:jdk21-unit

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
ci:report:2570942728

**Comment by julschong** [comment]
ci:report:2570911332

**Comment by julschong** [comment]
JDK21 Regression:
- com.liferay.portal.security.xml.SecureXMLFactoryProviderImplTest

Expected Regression on jdk21 when NewEnv is set properly (Will be fixed with CLDR PR after this PR is merged):
- com.liferay.portal.bootstrap.FrameworkRestartTest
- com.liferay.portal.search.web.internal.result.display.context.builder.SearchResultSummaryDisplayContextBuilderTest
- com.liferay.dynamic.data.mapping.form.field.type.internal.date.DateDDMFormFieldValueRendererTest
- com.liferay.portal.kernel.util.LocaleUtilTest

**Comment by julschong** [comment]
SecureXMLFactoryProviderImplTest is out of memory, will change the java options to allow more.

File: portal-kernel/test/unit/com/liferay/portal/kernel/test/rule/NewEnvJVMTestRuleTest.java [hunk 1]
```diff
@@ -36,6 +39,12 @@ public class NewEnvJVMTestRuleTest {
 	public static void setUpClass() {
 		System.setProperty(
 			_SYSTEM_PROPERTY_KEY_ENVIRONMENT, _toString(_getEnvironment()));
+
+		System.setProperty(
+			_PARENT_RUNTIME_JAVA_HOME_KEY, System.getProperty("java.home"));
+
+		System.setProperty(
```
Comment (tinatian) [inline portal-kernel/test/unit/com/liferay/portal/kernel/test/rule/NewEnvJVMTestRuleTest.java:46]: @julschong Why you need to do this? env should be covered by 
```
System.setProperty(
			_SYSTEM_PROPERTY_KEY_ENVIRONMENT, _toString(_getEnvironment()));
```

**Comment by julschong** [comment]
pushed changes first, not ready to review yet

**Comment by julschong** [comment]
ci:test:jdk21-unit

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
see #7113

**Pull request closed without merging**